File: toys/misc/matlab/signals.m
#!/usr/local/bin/octave -qf

function do_square()
    f = 10;
    overSampRate = 30;
    fs = overSampRate*f;
    ncy = 5;
    t = 0:1/fs:ncy*1/f-1/fs;
    g = sign(sin(2*pi*f*t));
    plot(t,g); title(['Square Wave f =' num2str(f), 'Hz']);
    pause();
end

function do_sine()
    f = 10;
    overSampRate = 30;
    fs = overSampRate*f;
    phase = 1/3*pi;
    ncy = 5;
    t = 0:1/fs:ncy*1/f-1/fs;
    g = sin(2*pi*f*t+phase);
    plot(t,g); title(['Sine Wave f =' num2str(f), 'Hz']);
    pause();
end

%do_sine();
%do_square();

File: toys/misc/py/signals.py
#!/usr/bin/env python3

# https://thehackerdiary.wordpress.com/2017/06/09/it-is-ridiculously-easy-to-generate-any-audio-signal-using-python/

import struct
import numpy as np
from scipy import signal as sg
import matplotlib.pyplot as plt


sampling_rate = 44100
freq = 440
samples = 44100
x = np.arange(samples)

# sine wave
y = 100 * np.sin(2 * np.pi * freq * x / sampling_rate)

# square wave
y = 100 * sg.square(2 * np.pi * freq * x / sampling_rate)
# square wave with duty cycle
y = 100 * sg.square(2 * np.pi * freq * x / sampling_rate, duty = 0.8)
# sawtooth wave
y = 100 * sg.sawtooth(2 * np.pi * freq * x / sampling_rate)


f = open('test.wav','wb')
for i in y:
    f.write(struct.pack('b',int(i)))
f.close()

n = len(y)
Y = np.fft.fft(y)/n # fft computing and normalization    

plt.plot(x,Y)
plt.show()

